src/ROUTES/LOGINPAGE/Login.tsx

```diff
@@ -0,0 +1,148 @@
+import { useState, useEffect } from "react";
+import { useDispatch } from "react-redux/es/exports";
+import { setProfileDetails } from "../../REDUX/profileDetailSlice.js";
+import { useLogin } from "react-admin";
+//@ts-ignore
+import "./login.css";
+import { toast } from "react-toastify";
+import { Flip } from "react-toastify";
+
+export const Login = (props: any) => {
+  interface IDetails {
+    name: String;
+    password: String;
+  }
+  const name = localStorage.getItem("name");
+  const password = localStorage.getItem("password");
+
+  const [auth, setAuth] = useState<IDetails>({
+    name: name || "",
+    password: password || "",
+  });
+  const [msg, setMsg] = useState("");
+  const dispatch = useDispatch();
+  const dateAndTimeHandler = (tabledata: any) => {
+    const newTableData = tabledata?.audit?.map((data: any, idx: any) => {
+      if (typeof data.datetime === "number") {
+        let datetime = new Date(data.datetime).toLocaleDateString("en-GB");
+        datetime += " " + new Date(data.datetime).toLocaleTimeString();
+
+        return { ...data, no: data.realid, id: idx + 1, datetime };
+      }
+      return { ...data, no: data.realid, id: idx + 1 };
+    });
+    if (newTableData) tabledata.audit = newTableData;
+    props.setFinalTableData(tabledata);
+    localStorage.setItem("tabledata", JSON.stringify(tabledata));
+  };
+
+  const inputHandler = (e: any) => {
+    if (msg) setMsg("");
+    const { name, value } = e.target;
+    setAuth({ ...auth, [name]: value });
+  };
+
+  const login = useLogin();
+  useEffect(() => {
+    localStorage.removeItem("name");
+    localStorage.removeItem("password");
+    setAuth({ name: "", password: "" });
+  }, []);
+  //@ts-ignore
+  const submitHandler = (e) => {
+    e.preventDefault();
+    //@ts-ignore
+    const data = JSON.parse(localStorage.getItem("data"));
+    let i = 0;
+    //@ts-ignore
+    const tableData = JSON.parse(localStorage.getItem("tabledata")) || {};
```

Comment:
again, in my opinion, the login page should not be aware of how we have implemented the data storage.  It should only need to refer to the dataProvider, or the react-admin `use` hooks.